ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -268,6 +245,32 @@ namespace enigma
 
     instance_event_iterator = new inst_iter(NULL,NULL,NULL);
```

Comment:
Care to fix this memory leak, while you're at it?

Reply:
It depends, how complicated is it? I'll do it if you show me the correct way. I may put it into a separate pull request though.

Reply:
```cpp
static inst_iter sentinel_iterator(NULL, NULL, NULL);
instance_event_iterator = &sentinel_iterator;
```

Reply:
Several of us were responsible for it, this one was specifically forthevin though, but I continued to copy it into other places as I added additional features. Regardless, I've opened #1441 for your review and you should explain a little better exactly how it fixes a memory leak because I don't quite understand myself. That said, I don't want to discuss this any further on this pull request. Thanks.